System.IO.Abstractions/FileSystem.cs

```diff
@@ -30,26 +36,25 @@ public IFileStreamFactory FileStream
         PathBase path;
         public PathBase Path
         {
-            get { return path ?? (path = new PathWrapper()); }
+            get { return path ?? (path = new PathWrapper(this)); }
         }
 
         DirectoryInfoFactory directoryInfoFactory;
         public IDirectoryInfoFactory DirectoryInfo
         {
-            get { return directoryInfoFactory ?? (directoryInfoFactory = new DirectoryInfoFactory()); }
+            get { return directoryInfoFactory ?? (directoryInfoFactory = new DirectoryInfoFactory(this)); }
         }
 
-        private readonly Lazy<DriveInfoFactory> driveInfoFactory = new Lazy<DriveInfoFactory>(() => new DriveInfoFactory());
```

Comment:
I'm not sure why this is the only one done as a `Lazy<T>`, but I've left it as-is.
I can fix this up to make is all consistent in another PR if you like?
(Or do it as part of this one if you don't mind the slight diff noise.)

Reply:
Would be great to make this consistent (I slightly prefer `Lazy<T>`)

Reply:
I've binned all the lazy entirely and just construct these instead - given all the constructors do basic field init (added in this PR) and nothing else, I really can't understand why these were made lazy in the first place!